### PR TITLE
[FIX] hr_attendance : fix onboarding screen

### PR DESCRIPTION
**Pull request by mepe-odoo**
STEP TO REPRODUCE:
==================

    ISSUE 1:
    ========
        0- Create a database without demodata
        1- Go on Attendance Application
        2- Click on Load demo data
        3- Write in "gegegege" in search bar for employee
        4 - Click again on Load demo data
    An error will be raised. To fix that if the demo data have been already launched this button will be hide.

    ISSUE 2:
    ========
    To have this issue, today's month need to have less than 31 days.
        0- Create a database without demodata
        1- Go on Attendance Application
        2- Click on Load demo data
    An error about month's number will be raised.

task-3983778

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr_attendance/views/hr_attendance_view.xml [hunk 1]
```diff
@@ -258,41 +258,7 @@
...
-                    </div>
-                    <div class="row justify-content-center">
-                        <a type="action" class="btn btn-secondary d-block col-2" name="%(action_load_demo_data)d">
-                            Load sample data
-                        </a>
-                    </div>
-                </div>
-            </p>
+        <field name="help">
         </field>
```
Comment (Flotchet) [inline addons/hr_attendance/views/hr_attendance_view.xml:262]: you can just use an self closed tag

**Review by Flotchet — CHANGES_REQUESTED** [review]
hello smol change

File: addons/hr_attendance/static/src/views/attendance_helper_view.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        this.orm = useService("orm");
+        this.actionService = useService("action");
+        this.state = useState({
+            hasDemoData: false,
+        });
+        onWillStart(async () => {
+            this.employeeNumber = await this.orm.searchCount(
+                "hr.employee", []);
+            const demoEmployeesCount = await this.orm.search(
+                "hr.employee", [["category_ids.name", "=", "Demo"]]);
```
Comment (tivisse) [inline addons/hr_attendance/static/src/views/attendance_helper_view.js:18]: Ziziznotokwithtranslations, use ``ref`` instead.

File: addons/hr_attendance/static/src/views/attendance_helper_view.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    static template = "hr_attendance.AttendanceActionHelper";
+    static props = ["noContentHelp"];
+    setup() {
+        this.orm = useService("orm");
+        this.actionService = useService("action");
+        this.state = useState({
+            hasDemoData: false,
+        });
+        onWillStart(async () => {
+            this.employeeNumber = await this.orm.searchCount(
```
Comment (tivisse) [inline addons/hr_attendance/static/src/views/attendance_helper_view.js:15]: I don't like the fact we make 3 RPC calls to actually always hide a button for real customer databases.

Make an api.model method instead. :)

**Review by tivisse — CHANGES_REQUESTED** [review]
Small review.

File: addons/hr_attendance/models/hr_attendance.py [hunk 4]
```diff
@@ -439,20 +440,21 @@ def action_out_attendance_maps(self):
             'target': 'new'
         }
 
-    def _load_demo_data(self):
-        # Check if demo data was already installed
+    @api.model
+    def hasDemoData(self):
```
Comment (tivisse) [inline addons/hr_attendance/models/hr_attendance.py:444]: Ziziznotjscode

**Comment by tivisse** [comment]
@robodoo r+ rebase-ff

**Comment by tivisse** [comment]
@robodoo r+

**Pull request closed without merging**